### PR TITLE
Link back to top to title of spec

### PR DESCRIPTION
**Pull request by OrionStar25**
Fixes : #1545 
Since title is always present.

### CODE REVIEW

File: src/core/structure.js [hunk 1]
```diff
@@ -145,7 +145,7 @@ export function run(conf, doc, cb) {
     }
 
     var $link = $(
-      "<p role='navigation' id='back-to-top'><a href='#toc'><abbr title='Back to Top'>&uarr;</abbr></a></p>"
+      "<p role='navigation' id='back-to-top'><a href='#title'><abbr title='Back to Top'>&uarr;</abbr></a></p>"
```
Comment (marcoscaceres) [inline src/core/structure.js]: Ah, good choice 👌

**Comment by marcoscaceres** [comment]
Trying to decide if we should add a test for this... I’m thinking, check if this element was added, it links to #title, and that the spec has an element #title. That should catch any potential regressions. @OrionStar25, what do you think?

**Comment by OrionStar25** [comment]
According to the user guide
https://github.com/w3c/respec/wiki/ReSpec-Editor's-Guide

It says " the title of the document is reused as the title of the specification in the resulting document's h1. That way, they are always in sync and you need not worry about specifying it twice. Including a title is required."
I thought hence its always mandatory to specify a title and hence we wouldn't have a case when the #title element isn't present. Please tell me if I'm wrong!

**Comment by marcoscaceres** [comment]
You are definitely not wrong, but things/code changes. For example, we are reworking the title code in a separate PR:
 https://github.com/w3c/respec/pull/1516

So the test would assure that what you stated above remains demonstratively true. It’s a bit of insurance against our future selves:)

**Comment by OrionStar25** [comment]
Then all the more reason to go ahead with a test! I have a generic doubt though, when you mention "tests", would if-else statements count as test statements? or they always have to be something like toBeTruthy() ?

**Comment by marcoscaceres** [comment]
Yeah, they always have to be an assertion of some sort - if/else wouldn’t count.

**Comment by OrionStar25** [comment]
Alright then, could you please tell me where I should be adding tests for this @marcoscaceres ?

**Comment by marcoscaceres** [comment]
That would be the right place for tests. Note that we have a new async way of writing test, for example:

https://github.com/w3c/respec/blob/develop/tests/spec/core/best-practices-spec.js#L5

Makes them a little easier to work with.

**Comment by OrionStar25** [comment]
@marcoscaceres please review this.

File: tests/spec/core/structure-spec.js [hunk 2]
```diff
@@ -42,49 +42,72 @@ describe("Core - Structure", () => {
...
   });
+
+  it("should link to the title of the document", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body
+    };
+    const doc = await makeRSDoc(ops);
+    const title = doc.getElementById("title");
+    expect(title).toEqual(!null);
```
Comment (marcoscaceres) [inline tests/spec/core/structure-spec.js]: Right now, this is checking: element === true, which is not correct (hence it’s failing). We actually want `.toBeTruthy()` or, `.not.toEqual(null)`.

Comment (OrionStar25) [reply tests/spec/core/structure-spec.js]: Oh! I'm sorry, I shouldve checked the differences between these assertions before sending this. Somehoe my tests weren't showing any errors. I'll make the change.

**Comment by OrionStar25** [comment]
@marcoscaceres I changed it to .toBeTruthy()

File: tests/spec/core/structure-spec.js [hunk 3]
```diff
@@ -42,49 +42,72 @@ describe("Core - Structure", () => {
...
   });
+
+  it("should link to the title of the document", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body
+    };
+    const doc = await makeRSDoc(ops);
+    const title = doc.getElementById("title");
+    expect(title).toBeTruthy();
```
Comment (marcoscaceres) [inline tests/spec/core/structure-spec.js]: Ok, now we need to check that the actual element was added and that it links to the title.

Comment (OrionStar25) [reply tests/spec/core/structure-spec.js]: Alright, just to make sure Im understanding this right,
this test confirmed that a title element is present. The next assertion to be written would be that that arrow should link to our title. 
Is that right?

Comment (marcoscaceres) [reply tests/spec/core/structure-spec.js]: Exactly right. Remember, the goal of the tests is to make sure it exercises every line of code you wrote - so always double check each new line is being run.

**Comment by OrionStar25** [comment]
I wasn't quite sure what to do after selecting the link so I just checked if it exists. :sweat_smile:

**Comment by OrionStar25** [comment]
Oh, alright! I should remember to rebate my branch often, sorry about that. Is there anything left to do here?

**Comment by marcoscaceres** [comment]
Nope all done! Nice work, @OrionStar25. Just waiting for TravisCI to run the tests and let us know it's ok... Travis is being very slllloooooooowwww today.

**Comment by OrionStar25** [comment]
And Travis build failed 😭

**Comment by marcoscaceres** [comment]
Thankfully, not related to your code (see logs). I’ll restart it and should be fine 👌

**Comment by marcoscaceres** [comment]
Restarted the build.

**Comment by OrionStar25** [comment]
I'm out of the city for 3 days. And logs don't show on mobile. I'll work on it, if it needs modification,  once I get back for sure 🤘

**Comment by marcoscaceres** [comment]
Ok, going to take this in for now.

**Comment by OrionStar25** [comment]
@marcoscaceres should anything be done here about the build error? I can't seem to understand this. Also, its still working the same way (linking back to #toc).

**Comment by marcoscaceres** [comment]
It's working perfectly and was deployed a few days ago 😃 

You might have a cached version of the spec. Check:
https://w3c.github.io/payment-request/#informative-references

You can see it's all in order: 
screenshot 2018-03-14 11 21 13

**Pull request merged**